containers/jersey2-routing/src/main/java/com/yahoo/gondola/container/RoutingFilter.java

```diff
@@ -488,12 +497,12 @@ private void waitSynced(String clusterId) {
             try {
                 Thread.sleep(500);
                 long now = System.currentTimeMillis();
-                int diff = gondola.getCluster(clusterId).getLastSavedIndex() - routingHelper.getAppliedIndex(clusterId);
-                if (checkTime - now > 10000) {
+                int diff = gondola.getCluster(clusterId).getCommitIndex() - routingHelper.getAppliedIndex(clusterId);
+                if (now - checkTime > 10000) {
                     checkTime = now;
-                    logger.warn("Recovery running for {} seconds, {} logs left", now - startTime, diff);
+                    logger.warn("Recovery running for {} seconds, {} logs left", (now - startTime) / 1000, diff);
                 }
-                synced = diff == 0;
+                synced = diff <= 0;
```

Comment:
Note, in the beginning, commitIndex = 0, and appliedIndex = 1. It should be an existing bug in gondola since commitIndex should never be 0.